cpp/src/holtwinters/holtwinters.hpp

```diff
@@ -0,0 +1,127 @@
+                                       int frequency, int start_periods,
+                                       Dtype *level, Dtype *trend,
+                                       Dtype *season,
+                                       aion::SeasonalType seasonal);
+template <typename Dtype>
+aion::AionStatus stl_decomposition_gpu(const Dtype *ts, int n, int batch_size,
+                                       int frequency, int start_periods,
+                                       Dtype *level, Dtype *trend,
+                                       Dtype *season,
+                                       aion::SeasonalType seasonal);
+
+//holtwinters_eval.hpp
+template <typename Dtype>
+void holtwinters_eval_cpu(const Dtype *ts, int n, int batch_size, int frequency,
+                          const Dtype *start_level, const Dtype *start_trend,
+                          const Dtype *start_season, const Dtype *alpha,
+                          const Dtype *beta, const Dtype *gamma, Dtype *level,
+                          Dtype *trend, Dtype *season, Dtype *xhat,
+                          Dtype *error, aion::SeasonalType seasonal);
+template <typename Dtype>
+void holtwinters_eval_gpu(const Dtype *ts, int n, int batch_size, int frequency,
+                          const Dtype *start_level, const Dtype *start_trend,
+                          const Dtype *start_season, const Dtype *alpha,
+                          const Dtype *beta, const Dtype *gamma, Dtype *level,
+                          Dtype *trend, Dtype *season, Dtype *xhat,
+                          Dtype *error, aion::SeasonalType seasonal);
+
+template <typename Dtype>
+Dtype holtwinters_eval_host(int id, const Dtype *ts, int n, int batch_size,
+                            int frequency, int shift, Dtype plevel,
+                            Dtype ptrend, const Dtype *start_season,
+                            const Dtype *beta, const Dtype *gamma, Dtype alpha_,
+                            Dtype beta_, Dtype gamma_, Dtype *level,
+                            Dtype *trend, Dtype *season, Dtype *xhat,
+                            aion::SeasonalType seasonal);
+template <typename Dtype, bool additive_seasonal>
+__device__ Dtype holtwinters_eval_device(
+  int tid, const Dtype *ts, int n, int batch_size, int frequency, int shift,
+  Dtype plevel, Dtype ptrend, Dtype *pseason, int pseason_width,
+  const Dtype *start_season, const Dtype *beta, const Dtype *gamma,
+  Dtype alpha_, Dtype beta_, Dtype gamma_, Dtype *level, Dtype *trend,
+  Dtype *season, Dtype *xhat);
+
+template <typename Dtype>
+Dtype bound_host(Dtype val, Dtype min = .0, Dtype max = 1.);
+
+template <typename Dtype>
+__device__ Dtype abs_device(Dtype val);
+template <typename Dtype>
+__device__ Dtype bound_device(Dtype val, Dtype min = .0, Dtype max = 1.);
+
+//holtwinters_forecast.hpp
+template <typename Dtype>
+void holtwinters_forecast_cpu(Dtype *forecast, int h, int batch_size,
```

Comment:
Why are we exposing a separate cpu based fit/predict methods on our interface?

Reply:
Why not just only expose methods that run on GPUs, which is being followed in all our current algos?

Reply:
That's a good idea, can refactor to remove all CPU code